#!/usr/bin/env python
PACKAGE = "auv_bringup"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Mission parameters
gen.add("pool_selection", str_t, 0, "Pool selection (pool_a or pool_d)", "pool_a")
gen.add("selected_animal", str_t, 0, "Selected animal for mission (shark or sawfish)", "sawfish")
gen.add("wall_reference_yaw", double_t, 0, "Yaw angle of the wall reference frame", 0.0, -3.14, 3.14)
gen.add("slalom_direction", str_t, 0, "Slalom direction", "right")

# Pool A configuration values
gen.add("pool_a_gate_exit_angle", double_t, 0, "Pool A - Gate exit angle in degrees", 90.0, -180.0, 180.0)
gen.add("pool_a_slalom_exit_angle", double_t, 0, "Pool A - Slalom exit angle in degrees", 50.0, -180.0, 180.0)
gen.add("pool_a_bin_exit_angle", double_t, 0, "Pool A - Bin exit angle in degrees", 0.0, -180.0, 180.0)
gen.add("pool_a_torpedo_exit_angle", double_t, 0, "Pool A - Torpedo exit angle in degrees", 180.0, -180.0, 180.0)

# Pool D configuration values
gen.add("pool_d_gate_exit_angle", double_t, 0, "Pool D - Gate exit angle in degrees", -90.0, -180.0, 180.0)
gen.add("pool_d_slalom_exit_angle", double_t, 0, "Pool D - Slalom exit angle in degrees", -50.0, -180.0, 180.0)
gen.add("pool_d_bin_exit_angle", double_t, 0, "Pool D - Bin exit angle in degrees", 0.0, -180.0, 180.0)
gen.add("pool_d_torpedo_exit_angle", double_t, 0, "Pool D - Torpedo exit angle in degrees", 0.0, -180.0, 180.0)

# Current active exit angles (automatically updated based on pool selection)
gen.add("gate_exit_angle", double_t, 0, "Active gate exit angle in degrees", 0.0, -180.0, 180.0)
gen.add("slalom_exit_angle", double_t, 0, "Active slalom exit angle in degrees", 0.0, -180.0, 180.0)
gen.add("bin_exit_angle", double_t, 0, "Active bin exit angle in degrees", 0.0, -180.0, 180.0)
gen.add("torpedo_exit_angle", double_t, 0, "Active torpedo exit angle in degrees", 0.0, -180.0, 180.0)

exit(gen.generate(PACKAGE, "auv_bringup", "SmachParameters"))
